src/main/scala/com/daml/projection/javadsl/BatchSource.scala

```diff
@@ -33,18 +38,29 @@ object BatchSource {
   /**
    * Creates a [[BatchSource]] from existing batches, useful for testing purposes.
    */
-  def create[E](batches: java.lang.Iterable[Batch[E]]): BatchSource[E] = S.BatchSource(batches.asScala.toList).toJava
+  def create[E](
+      batches: java.lang.Iterable[Batch[E]],
+      templateIdFromEvent: java.util.function.Function[E, ju.Optional[JIdentifier]],
```

Comment:
```suggestion
      identifierFromEvent: java.util.function.Function[E, ju.Optional[JIdentifier]],
```

Reply:
will change it to contractTypeIdFromEvent